sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -628,12 +628,10 @@ def _cleanup(self, on_exit=False):
           '  with Pipeline() as p:\n'
           '    p.apply(..)\n'
           'This ensures that the pipeline finishes before this program exits.')
-    has_exception = None
     for callback in self._cleanup_callbacks:
       try:
         callback()
-      except Exception:
-        has_exception = True
-    self._cleanup_callbacks = ()
-    if has_exception:
-      raise
+      except Exception as e:
+        raise e
+      finally:
+        self._cleanup_callbacks = ()
```

Comment:
`self._cleanup_callbacks = ()` executes whether there's an exception or not.

Reply:
this code has slightly different behavior, right? The new code may not call all of the cleanup callbacks. If one of them errors out before the end, then the final callbacks will not be processed, right?

Maybe you can store all exceptions, and raise a RuntimeException with information about all of them?

Reply:
Done 👍 , not sure if the format is correct but it does what you suggest.